pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -167,6 +170,14 @@ type AMISelectorTerm struct {
 	Owner string `json:"owner,omitempty"`
 }
 
+// PlacementGroupSelectorTerm defines the selection logic for ec2 placement groups
+// that are used to launch nodes. If multiple fields are used for selection, the requirements are ANDed
+type PlacementGroupSelectorTerm struct {
```

Comment:
There's a question here of how we should pass requirements down when using something like `cluster` placement groups OR when using something like `spread` with `rack`. 

When using the `spread` with `rack`, the EC2 documentation mentions that you can't launch more than 7 instances in a single AZ (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#:~:text=A%20rack%20spread%20placement%20group%20supports%20a%20maximum%20of%20seven%20running%20instances%20per%20Availability%20Zone), which means that we are restricted with our node limits when using this type of placement group. One option here is that we could constrain our requirements when using this type of placement group so that when we run out of instances to launch in a single AZ, we just get rid of that AZ from our requirements and start launching in other AZs. Another option here is that we allow to select on multiple placement groups and if we run out of space with one of the `spread` placement groups we just move to the other one.

This handle differently, though, when we are dealing with `cluster` or `partition` placement groups where there is no limit on the number of instances that can be launched and it makes less sense to allow for multiple placement groups to be selected on. 

Based on all of these considerations, I think we should think about writing up a small design that considers the use-cases around placement groups, how they meld with requirements and what the API surface should look like.